src/telliot_feeds/sources/mimicry/mimicry_mashup_source.py

```diff
@@ -0,0 +1,207 @@
+        """
+        if not self.collections:
+            logger.info("No NFTGo collections specified.")
+            return None
+
+        nftgo_url = "https://data-api.nftgo.io/eth/v1/collection/{contract_address}/metrics"
+        headers = {"X-API-KEY": self.api_key, "accept": "application/json"}
+
+        urls = []
+        for item in self.collections:
+            try:
+                contract_address = item[1]
+            except IndexError as e:
+                logger.warning(f"Specified NFTGo collection is not valid: {e}.")
+                return None
+            urls.append(nftgo_url.format(contract_address=contract_address))
+
+        responses = await self.fetch_urls(urls=urls, headers=headers)
+        collection_mcaps = []
+        metric = f"{self.metric}_{self.currency}".replace("-", "_")
+        for response in responses:
+            if response is not None:
+                try:
+                    market_cap = response[metric]
+                    collection_mcaps.append(market_cap)
+                except KeyError as e:
+                    logger.warning(f"Failed to parse NFTGo response, keyerror: {e}.")
+                    return None
+            else:
+                return None
+
+        return sum(collection_mcaps)
+
+    async def fetch_tokens_mcap(self) -> Optional[Any]:
+        """
+        Fetch list of tokens' market cap from coingecko api.
+
+        Returns:
+            Sum of all tokens' market cap, or None if the request fails.
+
+        """
+        if not self.tokens or not self.currency:
+            logger.info("No tokens specified, check query.")
+            return None
+        coingecko_url = "https://api.coingecko.com/api/v3/coins/{network}/contract/{contract_address}"
+        urls = []
+        for item in self.tokens:
+            try:
+                network = block_chains[item[0]]
+                contract_address = item[2]
+            except Exception as e:
+                logger.warning(f"Specified CoinGecko token is not valid: {e}.")
+                return None
+
+            urls.append(coingecko_url.format(network=network, contract_address=contract_address))
+
+        responses = await self.fetch_urls(urls=urls)
+        token_mcaps = []
+        for response in responses:
+            if response is not None:
+                try:
+                    market_cap = response["market_data"]["market_cap"][self.currency.lower()]
+                    token_mcaps.append(market_cap)
+                except Exception as e:
+                    logger.warning(f"Failed to parse CoinGecko response: {e}.")
+                    return None
+            else:
+                return None
+
+        return sum(token_mcaps)
+
+    async def fetch_new_datapoint(self) -> OptionalDataPoint[Any]:
+        """Fetch new data point from NFTGo API."""
+        collections_mcaps = await self.fetch_collections_mcap()
+        if collections_mcaps is None:
```

Comment:
Can refactor into one `if` like:
```py
collections_mcaps = ...
tokens_mcaps = await self.fetch_tokens_mcap()
 if tokens_mcaps is None or collections_mcaps is None:
            logger.warning(f"Failed to fetch all market caps. tokens = {tokens_mcaps}, collections = {collections_mcaps")
            return None, None

Reply:
I was trying to avoid making a call if the previous call failed. but will change